mlflow/tensorflow.py

```diff
@@ -255,17 +280,30 @@ def _load_tensorflow_saved_model(tf_saved_model_dir, tf_sess, tf_meta_graph_tags
                                  signature definition mapping. For more information, see the
                                  ``signature_def_map`` parameter of the
                                  ``tf.saved_model.builder.SavedModelBuilder`` method.
-    :return: A TensorFlow signature definition of type:
+    :param tf_sess: The TensorFlow session in which to load the metagraph.
+                    Required in TensorFlow versions < 2.0.0. Unused in TensorFlow versions >= 2.0.0
+    :return: For TensorFlow versions < 2.0.0:
+             A TensorFlow signature definition of type:
              ``tensorflow.core.protobuf.meta_graph_pb2.SignatureDef``. This defines input and
              output tensors within the specified metagraph for inference.
+             For TensorFlow versions >= 2.0.0:
+             A callable graph (tensorflow.function) that takes inputs and returns inferences.
     """
-    meta_graph_def = tensorflow.saved_model.loader.load(
+    if LooseVersion(tensorflow.__version__) < LooseVersion('2.0.0'):
+        meta_graph_def = tensorflow.saved_model.loader.load(
             sess=tf_sess,
             tags=tf_meta_graph_tags,
             export_dir=tf_saved_model_dir)
-    if tf_signature_def_key not in meta_graph_def.signature_def:
-        raise MlflowException("Could not find signature def key %s" % tf_signature_def_key)
-    return meta_graph_def.signature_def[tf_signature_def_key]
+        if tf_signature_def_key not in meta_graph_def.signature_def:
+            raise MlflowException("Could not find signature def key %s" % tf_signature_def_key)
```

Comment:
I know the original code didn't do it, but maybe we can update this message to include the list of available keys (e.g. `list(meta_graph_def.signature_def.keys())`)? We could also deduplicate this check with the check for 2.0 so that we only have to do it once (the last step in both TF 1.x & 2.0 cases is returning a value from a dict, so we could use the same code path for both)